inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/tasks/TrainingTask.java

```diff
@@ -94,132 +94,111 @@ public void run()
             }
         };
         
-        boolean seenSuccessfulTraining = false;
```

Comment:
This change needs definitely some verification with tests.

Reply:
In this modification, in order to reduce the number of loops, I used a way of traversing the array to split a large loop into two small loops. This has a limited performance improvement, which greatly improves the comprehensibility of the code, so I don't think it makes sense to change the comprehensibility of the code here.